### PR TITLE
Fix typo method and class. Deprecate the old ones.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheUpdateHandler.java [hunk 1]
```diff
@@ -65,6 +65,25 @@
     /**
      * Creates a cache entry for the given request, origin response message and response content.
      */
+    public HttpCacheEntry createCacheEntry(
```
Comment (ok2c) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CacheUpdateHandler.java:68]: @arturobernalg `CacheUpdateHandler` is not a part of public API. Deprecation is not needed  here.

File: httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingExecBase.java [hunk 2]
```diff
@@ -139,6 +139,23 @@ public long getCacheUpdates() {
         return cacheUpdates.get();
     }
 
+    /**
```
Comment (ok2c) [inline httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingExecBase.java:142]: @arturobernalg Same here. This is not a public method.

File: httpclient5/src/main/java/org/apache/hc/client5/http/async/methods/IgnoreCompleteExceptonFutureCallback.java [hunk 3]
```diff
@@ -30,6 +30,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
```
Comment (ok2c) [inline httpclient5/src/main/java/org/apache/hc/client5/http/async/methods/IgnoreCompleteExceptonFutureCallback.java:33]: @arturobernalg Please make this class extend `IgnoreCompleteExceptionFutureCallback`.

**Review by ok2c — CHANGES_REQUESTED** [review]

File: httpclient5/src/main/java/org/apache/hc/client5/http/async/methods/IgnoreCompleteExceptonFutureCallback.java (hunk 3 above)
Comment (arturobernalg) [reply httpclient5/src/main/java/org/apache/hc/client5/http/async/methods/IgnoreCompleteExceptonFutureCallback.java:31]: > @arturobernalg Please make this class extend `IgnoreCompleteExceptionFutureCallback`.

Fixed al the remarks.
TY

**Review by ok2c — APPROVED** [review]

**Pull request merged**